src/pcre2_context.c

```diff
@@ -409,6 +410,35 @@ ccontext->stack_guard_data = user_data;
 return 0;
 }
 
+static uint16_t optimize_setbits[] = {
+  OPTIM_SETBITS
+};
+static uint16_t optimize_clearbits[] = {
+  OPTIM_CLEARBITS
+};
+
+PCRE2_EXP_DEFN int PCRE2_CALL_CONVENTION
+pcre2_set_optimize(pcre2_compile_context *ccontext, uint32_t directive)
+{
+/* We are reusing some symbolic constants which were already defined before
+ * pcre2_set_optimize() was added; convert to indexes in conversion tables */
+if (directive == PCRE2_NO_AUTO_POSSESS)
+  directive = 3;
+else if (directive == PCRE2_NO_DOTSTAR_ANCHOR)
+  directive = 5;
+else if (directive == PCRE2_NO_START_OPTIMIZE)
+  directive = 7;
+
+if (directive < (sizeof(optimize_setbits)/sizeof(uint16_t)))
+  {
+  /* Both of the public directive → private flag conversion tables should
+   * be the same size, but let's make sure */
+  PCRE2_ASSERT(directive < (sizeof(optimize_clearbits)/sizeof(uint16_t)));
```

Comment:
instead of a runtime assert, shouldn't this be triggering an `#error`?

Reply:
A simple implementation proposal in alexdowad/pcre2#5

Reply:
Looks good, applied. Thank you.